api/v1beta2/cryostat_types.go

```diff
@@ -758,6 +762,46 @@ type DatabaseOptions struct {
 	SecretName *string `json:"secretName,omitempty"`
 }
 
+// ObjectStorageOptions provides configuration options to the Cryostat application's object storage.
+type ObjectStorageOptions struct {
+	// Name of the secret containing the object storage secret access key. This secret must contain a
+	// ACCESS_KEY secret which is the object storage access key ID, and a SECRET_KEY secret which is the object storage secret access key.
+	// If using an external S3 provider requiring authentication then this must be provided.
+	// It is recommended that the secret should be marked as immutable to avoid accidental changes to secret's data.
+	// More details: [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable)
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	SecretName *string `json:"secretName,omitempty"`
+	// Configuration for external object storage providers.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Object Storage Provider Options"
+	Provider *ObjectStorageProviderOptions `json:"provider,omitempty"`
+
+	// TODO add configuration to override default bucket names
+}
+
+// ObjectStorageProviderOptions provides configuration options to the Cryostat application's external object storage.
+type ObjectStorageProviderOptions struct {
+	// The complete URL (not including authentication information) to the external object storage provider.
+	URL *string `json:"url,omitempty"`
+	// Whether path-style access should be used, as opposed to subdomain access. Defaults to true for compatibility.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Use Path-Style Access",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	UsePathStyleAccess *bool `json:"usePathStyleAccess,omitempty"`
+	// The object storage provider region.
+	Region *string `json:"region,omitempty"`
+	// Whether Cryostat should trust all TLS certificates presented by the external object storage provider. Defaults to false.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="TLS Trust All",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	TLSTrustAll *bool `json:"tlsTrustAll,omitempty"`
+	// The strategy Cryostat will use for storing files' metadata. The default 'tagging' strategy stores all metadata as object Tags.
+	// The 'metadata' strategy stores metadata as object Metadata, which is immutable but allows for more entries than Tags.
+	// The 'bucket' strategy stores metadata as separate files (ex. JSON object maps) in a dedicated bucket,
+	// with prefixes to differentiate the kind of object the metadata belongs to.
+	// +optional
```

Comment:
```suggestion
	// +optional
	// +kubebuilder:validation:Enum=tagging;metadata;bucket
	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:tagging","urn:alm:descriptor:com.tectonic.ui:select:metadata","urn:alm:descriptor:com.tectonic.ui:select:bucket"}
```

The kubebuilder marker adds server-side validation:
`The Cryostat "cryostat-sample" is invalid: spec.objectStorageOptions.provider.metadataMode: Unsupported value: "pineapple": supported values: "tagging", "metadata", "bucket"`

The operator-sdk marker adds a dropdown in the OpenShift UI:
<img width="3164" height="1514" alt="Screenshot 2025-09-24 at 13-46-44 Create Cryostat · Red Hat OpenShift Service on AWS" src="https://github.com/user-attachments/assets/7be2f9ad-09dc-4135-acca-e26d5b8814b4" />